### ISSUE TITLE
Astropy 3.2.2 test failures with Numpy 1.14.6

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#9388]
Some of the tests for Astropy 3.2.2 are failing with Numpy 1.14.6 which is causing the wheels to not be built for Python 3.7 (see https://github.com/astropy/astropy/issues/9371 for example). To reproduce with Python 3.7:

```
python -m venv python/debug-wheel
source ~/python/debug-wheel/bin/activate
pip install numpy==1.14.6
pip install astropy --no-binary :all:
pip install pytest-astropy
python -c "import astropy;astropy.test(args='-k test_regression_futuretimes_4302')"```
```

The error is then:

```
__________________________________________________________ test_regression_futuretimes_4302 ___________________________________________________________

    def test_regression_futuretimes_4302():
        """
        Checks that an error is not raised for future times not covered by IERS
        tables (at least in a simple transform like CIRS->ITRS that simply requires
        the UTC<->UT1 conversion).
    
        Relevant comment: https://github.com/astropy/astropy/pull/4302#discussion_r44836531
        """
        from astropy.utils.exceptions import AstropyWarning
    
        # this is an ugly hack to get the warning to show up even if it has already
        # appeared
        from astropy.coordinates.builtin_frames import utils
        if hasattr(utils, '__warningregistry__'):
            utils.__warningregistry__.clear()
    
        with catch_warnings() as found_warnings:
            future_time = Time('2511-5-1')
            c = CIRS(1*u.deg, 2*u.deg, obstime=future_time)
>           c.transform_to(ITRS(obstime=future_time))

python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/tests/test_regression.py:226: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/baseframe.py:1202: in transform_to
    return trans(self, new_frame)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/transformations.py:1377: in __call__
    curr_coord = t(curr_coord, curr_toframe)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/transformations.py:980: in __call__
    return supcall(fromcoord, toframe)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py:79: in cirs_to_itrs
    pmat = cirs_to_itrs_mat(itrs_frame.obstime)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py:31: in cirs_to_itrs_mat
    xp, yp = get_polar_motion(time)
python/debug-wheel/lib/python3.7/site-packages/astropy/coordinates/builtin_frames/utils.py:41: in get_polar_motion
    xp, yp, status = iers.IERS_Auto.open().pm_xy(time, return_status=True)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:660: in open
    cls.iers_table = cls.read(file=filename)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:507: in read
    table = cls._combine_a_b_columns(iers_a)
python/debug-wheel/lib/python3.7/site-packages/astropy/utils/iers/iers.py:469: in _combine_a_b_columns
    is_predictive = (table['UT1Flag_A'] == 'P') | (table['PolPMFlag_A'] == 'P')
python/debug-wheel/lib/python3.7/site-packages/astropy/table/column.py:986: in _compare
    return result.data if isinstance(result, Column) else result
python/debug-wheel/lib/python3.7/site-packages/astropy/table/column.py:1278: in data
    out.fill_value = self.fill_value
python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:3375: in __setattr__
    super(MaskedArray, self).__setattr__(attr, value)
python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:3659: in set_fill_value
    target = _check_fill_value(value, self.dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

fill_value = 'N', ndtype = dtype('bool')

    def _check_fill_value(fill_value, ndtype):
        """
        Private function validating the given `fill_value` for the given dtype.
    
        If fill_value is None, it is set to the default corresponding to the dtype.
    
        If fill_value is not None, its value is forced to the given dtype.
    
        The result is always a 0d array.
        """
        ndtype = np.dtype(ndtype)
        fields = ndtype.fields
        if fill_value is None:
            fill_value = default_fill_value(ndtype)
        elif fields:
            fdtype = [(_[0], _[1]) for _ in ndtype.descr]
            if isinstance(fill_value, (ndarray, np.void)):
                try:
                    fill_value = np.array(fill_value, copy=False, dtype=fdtype)
                except ValueError:
                    err_msg = "Unable to transform %s to dtype %s"
                    raise ValueError(err_msg % (fill_value, fdtype))
            else:
                fill_value = np.asarray(fill_value, dtype=object)
                fill_value = np.array(_recursive_set_fill_value(fill_value, ndtype),
                                      dtype=ndtype)
        else:
            if isinstance(fill_value, basestring) and (ndtype.char not in 'OSVU'):
                err_msg = "Cannot set fill value of string with array of dtype %s"
>               raise TypeError(err_msg % ndtype)
E               TypeError: Cannot set fill value of string with array of dtype bool

python/debug-wheel/lib/python3.7/site-packages/numpy/ma/core.py:473: TypeError
```

I have bisected this and it looks like this comes from https://github.com/astropy/astropy/pull/8904.

@mhvk, since this was your PR, is there any chance you could look into this? I know we aren't supporting Numpy 1.14 anymore in master, but it would be good to get all tests passing and do a last 3.2.x release before 4.0.

**Comment by mhvk** [comment]
Looks like there is a bug in `MaskedArray` that appeared in 1.14.6 and that breaks for subclasses. I see two options: just revert the PR (it is not such a big deal) or put back the old code for `MaskedArray` only:
```
if NUMPY_LT_1_15 and (isinstance(self, np.ma.MaskedArray) or
                      isinstance(other, np.ma.MaskedArray)):
    # On numpy 1.14.6, MaskedArray subclass comparison is broken.
    return getattr(self.data, op)(other)
```
Since `MaskedColumn` is not covered anyway by the test that I introduced, and since masked quanties work very poorly, this should be OK, even if ugly.

**Comment by astrofrog** [comment]
Let's maybe go with the second option, and I think this we can fix just in v3.2.x and leave master as-is, right? Do you have time to open a PR? (if not, I'll try later). Thanks for the quick reply!

**Comment by mhvk** [comment]
See #9389

**Comment by bsipocz** [comment]
This has been  fixed in 3.2.3
